service/history/queues/rescheduler.go

```diff
@@ -154,6 +159,32 @@ func (r *reschedulerImpl) Add(
 	}
 }
 
+func (r *reschedulerImpl) Reschedule(
+	namespaceIDs map[string]struct{},
+) {
+	r.Lock()
+	defer r.Unlock()
+
+	now := r.timeSource.Now()
+	for key, pq := range r.pqMap {
+		if _, ok := namespaceIDs[key.NamespaceID]; !ok {
+			continue
+		}
+
+		// set reschedule time for all tasks in this pq to be now
+		newPQ := r.newPriorityQueue()
+		for !pq.IsEmpty() {
+			rescheduled := pq.Remove()
+			rescheduled.rescheduleTime = now
+			newPQ.Add(rescheduled)
+		}
```

Comment:
Will update the logic here to use `NewPriorityQueueWithItems` after [this PR](https://github.com/temporalio/temporal/pull/3275#discussion_r958928189) is landed